AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/lro_spec.rb

```diff
@@ -8,221 +8,433 @@
+  end
+
+  it 'should fail for delete async retry failed' do
+    expect { @lros_client.delete_async_retry_failed().value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+
+  it 'should fail for delete async retry canceled' do
+    expect { @lros_client.delete_async_retrycanceled().value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+end
+
+describe 'Long Running Operation with retry' do
+  before(:all) do
+    base_url = ENV['StubServerURI']
+
+    dummyToken = 'dummy12321343423'
+    credentials = MsRest::TokenCredentials.new(dummyToken)
+
+    test_client = AutoRestLongRunningOperationTestService.new(credentials, base_url)
+    test_client.long_running_operation_retry_timeout = 0
+    @lroretrys_client = test_client.lroretrys
+    @product = Product.new
+    @product.location = "West US"
   end
 
   # Retryable errors
   it 'should retry PUT request on 500 responce' do
-    result = @client.lroretrys.put201creating_succeeded200(@product).value!
+    result = @lroretrys_client.put201creating_succeeded200(@product).value!
     expect(result.body.provisioning_state).to eq("Succeeded")
   end
 
   it 'should retry PUT request on 500 responce for async operation' do
-    result = @client.lroretrys.put_async_relative_retry_succeeded(@product).value!
+    result = @lroretrys_client.put_async_relative_retry_succeeded(@product).value!
     expect(result.body.provisioning_state).to eq("Succeeded")
   end
 
   it 'should retry DELETE request for provisioning status' do
-    result = @client.lroretrys.delete_provisioning202accepted200succeeded().value!
+    result = @lroretrys_client.delete_provisioning202accepted200succeeded().value!
     expect(result.response.status).to eq(200)
   end
 
   it 'should retry DELETE request on 500 responce' do
-    result = @client.lroretrys.delete202retry200().value!
+    result = @lroretrys_client.delete202retry200().value!
     expect(result.response.status).to eq(200)
   end
 
   it 'should retry POST request on 500 responce' do
-    result = @client.lroretrys.post202retry200(@product).value!
+    result = @lroretrys_client.post202retry200(@product).value!
     expect(result.response.status).to eq(200)
   end
 
   it 'should retry POST request on 500 responce for async operation' do
-    result = @client.lroretrys.post_async_relative_retry_succeeded(@product).value!
+    result = @lroretrys_client.post_async_relative_retry_succeeded(@product).value!
     expect(result.response.status).to eq(200)
   end
 
+  it 'should retry on 500 server responce in PUT request' do
+    result = @lroretrys_client.put_async_relative_retry_succeeded(@product).value!
+    expect(result.body.provisioning_state).to eq("Succeeded")
+  end
+
+  it 'should retry on 500 server responce in DELETE request' do
+    result = @lroretrys_client.delete_async_relative_retry_succeeded().value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should serve async POST operation' do

```

Comment:
yes exactly same. removed it.

---
In reply to: [69024543](https://github.com/Azure/autorest/pull/1188#discussion_r69024543) [](ancestors = 69024543)